public/app/plugins/panel/table/TablePanel.tsx

```diff
@@ -13,21 +13,44 @@ export class TablePanel extends Component<Props> {
     super(props);
   }
 
-  onColumnResize = (field: Field, width: number) => {
-    const current = this.props.fieldConfig;
-    const matcherId = FieldMatcherID.byName;
-    const prop = 'width';
-    const overrides = current.overrides.filter(
-      o => o.matcher.id !== matcherId || o.matcher.options !== field.name || o.properties[0].id !== prop
-    );
+  onColumnResize = (fieldIndex: number, width: number) => {
+    const { fieldConfig, data } = this.props;
+    const { overrides } = fieldConfig;
+    const frame = data.series[this.getCurrentFrameIndex()];
 
-    overrides.push({
-      matcher: { id: matcherId, options: field.name },
-      properties: [{ id: prop, value: width }],
-    });
+    if (!frame) {
+      return;
+    }
+
+    const field = frame.fields[fieldIndex];
+    if (!field) {
+      return;
+    }
+
+    const fieldName = field.name;
+    const matcherId = FieldMatcherID.byName;
+    const propId = 'custom.width';
+
+    // look for existing override
+    const override = overrides.find(o => o.matcher.id === matcherId && o.matcher.options === fieldName);
+
+    if (override) {
+      // look for existing property
+      const property = override.properties.find(prop => prop.id === propId);
+      if (property) {
+        property.value = width;
```

Comment:
Can we mutate fieldConfigs like this?

Reply:
we can, but would be better if done immutably.